src/Grpc.AspNetCore.Server/ServerCallContextExtensions.cs

```diff
@@ -0,0 +1,37 @@
+﻿#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using Grpc.AspNetCore.Server.Internal;
+using Microsoft.AspNetCore.Http;
+
+namespace Grpc.Core
+{
+    /// <summary>
+    /// Extension methods for ServerCallContext.
+    /// </summary>
+    public static class ServerCallContextExtensions
+    {
+        /// <summary>
+        /// Retrieve the <see cref="HttpContext"/> from a <see cref="ServerCallContext"/> if possible.
+        /// </summary>
+        /// <param name="serverCallContext">The <see cref="ServerCallContext"/> to extract from.</param>
+        /// <returns>The extracted <see cref="HttpContext"/>. If it cannot be extracted, null will be returned.</returns>
+        public static HttpContext GetHttpContext(this ServerCallContext serverCallContext)
```

Comment:
nit: add a note that this is for power users (and recommend read-only access)?   trying to change stuff in the httpcontext while the grpc layer potentially accesses the same fields can easily break things.

Reply:
I'd add a remarks section and say that this method provides access to the underlying HttpContext and that HttpContext is only available when using gRPC hosted on ASP.NET Core.